cmd/chisel/cmd_debug_check_release_archives.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	pathsObservations, err := checkReleaseArchives(release, archives)
+	if err != nil {
+		return err
+	}
+
+	var issues []any
+	type pathConflict struct {
+		Issue        string        `yaml:"issue"`
+		Path         string        `yaml:"path"`
+		Observations []observation `yaml:"observations"`
+	}
+	var sortedPaths []string
+	for path := range pathsObservations {
+		sortedPaths = append(sortedPaths, path)
+	}
+	slices.Sort(sortedPaths)
+	for _, path := range sortedPaths {
+		observations := pathsObservations[path]
+		hasConflict := false
+		base := observations[0]
+		for _, observation := range observations {
+			if observation.Kind != base.Kind ||
+				observation.Mode != base.Mode ||
+				observation.Link != base.Link {
+				hasConflict = true
+				break
+			}
+		}
+		if hasConflict {
+			issues = append(issues, pathConflict{
+				// At this time, there is only one possible type of conflict,
+				// we do not need to check.
+				Issue:        "path-conflict",
+				Path:         path,
+				Observations: observations,
+			})
+		}
+	}
+
+	if len(issues) > 0 {
+		yb, err := yaml.Marshal(issues)
+		if err != nil {
+			return fmt.Errorf("internal error: cannot marshal issue list: %s", err)
+		}
+		fmt.Fprintf(Stdout, "%s", string(yb))
+		return errors.New("issues found in the release archives")
+	}
+
+	return nil
+}
+
+func checkReleaseArchives(release *setup.Release, archives map[string]archive.Archive) (map[string][]observation, error) {
+	var orderedPkgs []string
+	for packageName := range release.Packages {
+		orderedPkgs = append(orderedPkgs, packageName)
+	}
+	slices.Sort(orderedPkgs)
+	var orderedArchives []string
+	for archiveName := range archives {
+		orderedArchives = append(orderedArchives, archiveName)
+	}
+	slices.Sort(orderedArchives)
+
+	pathsObservations := map[string][]observation{}
+	for _, archiveName := range orderedArchives {
+		archive := archives[archiveName]
+		logf("Processing archive %s...", archiveName)
+		for _, pkgName := range orderedPkgs {
+			if !archive.Exists(pkgName) {
+				continue
+			}
+			pkgReader, _, err := archive.Fetch(pkgName)
+			if err != nil {
+				return nil, err
+			}
+			dataReader, err := deb.DataReader(pkgReader)
+			if err != nil {
+				return nil, err
+			}
+			tarReader := tar.NewReader(dataReader)
+			for {
+				tarHeader, err := tarReader.Next()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return nil, err
+				}
+
+				path, ok := sanitizeTarPath(tarHeader.Name)
+				if !ok {
+					continue
+				}
+				if tarHeader.FileInfo().Mode().IsRegular() {
+					continue
+				}
+
+				// Make paths uniform: while directories always end in '/',
+				// symlinks don't.
+				path = strings.TrimSuffix(path, "/")
+
+				observations := pathsObservations[path]
+				found := false
+				// We look for a previous observation that extracts the same
+				// content in terms of mode, link, etc. and we add the package
+				// to it. If there is none, we create a new one.
```

Comment:
Let's compute the observation value ahead of time here, and then delegate the finding to a helper function. Something along the lines of:
```
index, ok := findObservation(observation, observation)
if !ok {
    observations = append(observations, observation)
}
```
This is the same note as earlier in this PR: we have very long functions with no attempt to break down algorithms into nice small abstractions.